data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OrderedConsumerVerticle.java

```diff
@@ -29,31 +37,38 @@
 import java.util.Map;
 import java.util.Set;
 
+import static dev.knative.eventing.kafka.broker.core.utils.Logging.keyValue;
+
 public class OrderedConsumerVerticle extends BaseConsumerVerticle {
 
   private static final Logger logger = LoggerFactory.getLogger(OrderedConsumerVerticle.class);
-  private static final Duration POLLING_TIMEOUT = Duration.ofMillis(100);
 
-  // poll wait ms = (POLL_WAIT_RECORD_FACTOR * pendingRecords) / (partitions - POLL_WAIT_PARTITION_FACTOR * Math.sqrt(partitions))
-  // Each record takes 10 ms to be processed
-  private static final double POLL_WAIT_RECORD_FACTOR = 10;
-  // This is used to take in account how much the parallelism is effective
-  private static final double POLL_WAIT_PARTITION_FACTOR = 1 / (double) Runtime.getRuntime().availableProcessors();
-  private static final long MAX_POLL_WAIT = 10 * 1000;
-  private static final int MIN_POLL_WAIT = 100;
+  private static final long POLLING_MS = 200L;
+  private static final Duration POLLING_TIMEOUT = Duration.ofMillis(1000L);
 
   private final Map<TopicPartition, OrderedAsyncExecutor> recordDispatcherExecutors;
+  private final DataPlaneContract.Egress egress;
+  private final Bucket bucket;
 
-  private int pendingRecords;
-  private boolean stopPolling;
+  private boolean closed;
 
-  public OrderedConsumerVerticle(Initializer initializer, Set<String> topics) {
+  public OrderedConsumerVerticle(final DataPlaneContract.Egress egress,
+                                 final Initializer initializer,
+                                 final Set<String> topics,
+                                 final int maxPollRecords) {
     super(initializer, topics);
 
-    this.recordDispatcherExecutors = new HashMap<>();
+    final var vReplicas = Math.max(1, egress.getVReplicas());
+    final var tokens = maxPollRecords * vReplicas;
+    this.bucket = new LocalBucketBuilder()
+      .addLimit(Bandwidth.classic(tokens, Refill.greedy(tokens, Duration.ofSeconds(1))))
+      .withSynchronizationStrategy(SynchronizationStrategy.SYNCHRONIZED)
+      .withMillisecondPrecision()
+      .build();
```

Comment:
@aslom if we inject this duration here, we can mock it and use minutes for unit tests and have more reliable low scale tests.